src/riak_kv_compile_tab.erl

```diff
@@ -71,35 +100,75 @@ is_compiling(BucketType) ->
         compiling_state() | notfound.
 get_state(BucketType) when is_binary(BucketType) ->
     case dets:lookup(?TABLE, BucketType) of
-        [{_,_,_,State}] ->
+        [{_,_,_,_,State}] ->
```

Comment:
Does it make sense to add a pattern match against the 4-tuple version, do a `lager:error` and/or try to recover?

Former seems like an easy thing to do, at least, and I assume would avoid a process crash.